lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4434,17 +4434,40 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
     if (FPClass & fcZero) {
       // Create zero integer constant and check for equality with bitcasted to
       // int float value
+      auto SetUpCMPToZero = [&](SPIRVValue *BitCastToInt,
+                                bool IsPositive) -> SPIRVValue * {
+        APInt ZeroInt = APInt::getZero(BitSize);
+        if (IsPositive) {
+          auto *ZeroConst =
+              transValue(Constant::getIntegerValue(IntOpLLVMTy, ZeroInt), BB);
+          return BM->addCmpInst(OpIEqual, ResTy, BitCastToInt, ZeroConst, BB);
+        }
+        // Created 'negated' zero
+        ZeroInt.setSignBit();
```

Comment:
As I understood APInt uses 2's complement, which does not have negative zero.  Are you sure this is giving the correct result?

Reply:
[this line in the test](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2151/files#diff-6e15e8a39c298484201aa544ea77f67acd2acae0c091213a6e2c09089d2e0ea2R24) added after setting signed bit seems about right

Reply:
Right, it works because you're not doing any (2's complement) operations on the APInt.

Reply:
Hi @svenvh 

Thanks for the review.
Is the use of APInt agreeable in this scenario, or do we have to change the logic here?

Thanks

Reply:
This becomes a moot point once the suggestion from @LU-JOHN below has been applied.